File: app/src/main/java/name/boyle/chris/sgtpuzzles/ActivityWithLoadButton.java
package name.boyle.chris.sgtpuzzles;

import android.content.ActivityNotFoundException;
import android.content.Intent;

import androidx.activity.result.ActivityResultLauncher;
import androidx.activity.result.contract.ActivityResultContracts;
import androidx.appcompat.app.AppCompatActivity;

public abstract class ActivityWithLoadButton extends AppCompatActivity {

    private final ActivityResultLauncher<String[]> loadLauncher = registerForActivityResult(new ActivityResultContracts.OpenDocument(), uri -> {
        final Intent intent = new Intent(Intent.ACTION_VIEW, uri, ActivityWithLoadButton.this, GamePlay.class);
        startActivity(intent);
        overridePendingTransition(0, 0);
    });

    protected void loadGame() {
        try {
            loadLauncher.launch(new String[]{"text/*", "application/octet-stream"});
        } catch (ActivityNotFoundException e) {
            Utils.unlikelyBug(this, R.string.saf_missing_short);
        }
    }
}

File: app/src/main/kotlin/name/boyle/chris/sgtpuzzles/ActivityWithLoadButton.kt
package name.boyle.chris.sgtpuzzles

import android.content.ActivityNotFoundException
import android.content.Intent
import android.content.Intent.ACTION_VIEW
import android.net.Uri
import androidx.activity.result.contract.ActivityResultContracts
import name.boyle.chris.sgtpuzzles.NightModeHelper.ActivityWithNightMode

abstract class ActivityWithLoadButton : ActivityWithNightMode() {

    private val loadLauncher =
        registerForActivityResult(ActivityResultContracts.OpenDocument()) { uri: Uri? ->
            if (uri == null) return@registerForActivityResult
            startActivity(
                Intent(ACTION_VIEW, uri, this@ActivityWithLoadButton, GamePlay::class.java)
            )
        }

    protected fun loadGame() {
        try {
            loadLauncher.launch(arrayOf("text/*", "application/octet-stream"))
        } catch (_: ActivityNotFoundException) {
            Utils.unlikelyBug(this, R.string.saf_missing_short)
        }
    }
}